core/utils/event_types.py

```diff
@@ -0,0 +1,59 @@
+"""This file contains all the different event types used betweent the
+Communication Manager and the Optimizers."""
+
+from enum import Enum
+
+class RawEventTypes(Enum):
+    """
+    Dictionary of events that the Optimizer might receive from the Communication
+    Manager.
+
+    The event indicates what has occurred previously in the Communication Manager.
+
+    This is essentially a 'memory' for the Optimizer so these are all past.
+    i.e., a "TRAIN" event would contain auxiliary information about what
+    occurred, in the form of a DMLResults object.
+
+    Contains all the functionality of what the Communication Manager might have
+    previously done and therefore is not specific to any one Optimizer.
+    """
+    NEW_SESSION = "new_session"
+    NEW_WEIGHTS = "new_weights"
```

Comment:
Nitpick but should `NEW_SESSION` and `NEW_WEIGHTS` be capitalized too?

Reply:
Will do